File: sharedApp/src/commonMain/kotlin/com/otero/recipetoshop/domain/model/despensa/Alimento.kt
package com.otero.recipetoshop.domain.model.despensa

import com.otero.recipetoshop.domain.model.ListaCompra.Productos
import com.otero.recipetoshop.domain.util.TipoUnidad

data class Alimento(
    val id_cestaCompra: Int? = null,
    val id_receta: Int? = null,
    val id_alimento: Int? = null,
    val nombre: String,
    var cantidad: Int = 0,
    val tipoUnidad: TipoUnidad,
    var active: Boolean
)

fun Alimento.toProducto(): Productos.Producto{
    return Productos.Producto(
        id_producto = -1,
        id_cestaCompra = id_cestaCompra!!.toInt(),
        nombre = nombre,
        peso = cantidad.toFloat(),
        oferta = "",
        imagen_src = "",
        precio_texto = "",
        cantidad = cantidad,
        precio_numero = 0f,
        precio_peso = "",
        query = nombre.lowercase().trim(),
        tipoUnidad = tipoUnidad
    )
}

fun List<Alimento>.toProductos(): List<Productos.Producto>{
    return map { it.toProducto() }
}

File: Scraping/alimento.py
import json

class Alimento():
    nombre = ''
    imagen_src = ''
    precio_peso = ''
    precio = ''
    oferta = ''
    query = ''
    supermercado = ''
    
    def build_alimento(query, nombre, imagen_src, precio_peso, oferta, precio, supermercado):
        alimento = Alimento()
        alimento.query = query
        alimento.nombre = nombre
        alimento.imagen_src = imagen_src
        alimento.precio_peso = precio_peso
        alimento.oferta = oferta
        alimento.precio = precio
        alimento.supermercado = supermercado
        return alimento
    
    def alimento_JSON(self) -> str:
        result = json.dumps(
            {
                'query': self.query,
                'nombre': self.nombre,
                'imagen_src': self.imagen_src,
                'precio_peso': self.precio_peso,
                'precio': self.precio,
                'oferta': self.oferta,
                'supermercado': self.supermercado
            }
        )
        return result